### DISCUSSION TITLE
Cart property naming

### DISCUSSION THREAD

**Discussion by glennjacobs** [lunarphp/lunar#1042] [Feature Requests]
In 0.3 we're looking to standardise the naming of cart line properties along the lines of the following

```
$unitAmount - base price of product (could be tax inclusive or exclusive)
$quantity - how many we want
$baseAmount - base price x qty
$discount - discount for the line
$lineAmount - (base price x qty) - discount
$taxAmount - tax amount (either inclusive or not)
```

**Comment by wychoong** [comment]
can this go out with #871 in 0.2 so that don't need to look at Discount module again in next release? 🥹

**Comment by glennjacobs** [comment]
@alecritson thoughts?

**Comment by alecritson** [comment]
It's an API change so feels like it should be `0.3` but also if there are reasonable upgrade steps then I guess we can squeeze it in.

**Comment by glennjacobs** [comment]
@alecritson I'm tempted to say for this RC release only a handful of devs have used it so far, so in terms of interruption it's probably better to make the change now. 

Technically, we should be doing this stuff in beta and we've learnt that skipping straight to an RC isn't a good idea. So future releases will have a beta first.

**Comment by alecritson** [comment]
Given that there's already a few changes in the mentioned PR, I'll do this as a separate branch then PR into #871 so it's easy to see the changes

**Comment by glennjacobs** [comment]
@wychoong we've had a good internal discussion on this and I'm going to leave it for 0.3.

The reason is these updates very much relate to the functionality of allowing prices to be inclusive of tax. When we come to implement that, there's a reasonable chance we'll have not considered something and end up changing the properties yet again.

I don't believe it will be a big headache to change when 0.3 is released, just a few property names to change and we'll document it.

**Comment by wychoong** [comment]
@glennjacobs sure 😃
